ql/src/java/org/apache/hadoop/hive/ql/exec/repl/atlas/AtlasProcess.java

```diff
@@ -0,0 +1,90 @@
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.ql.parse.SemanticException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Base class for Atlas Processes, viz. Export & Import
+ */
+public abstract class AtlasProcess {
+  private static final String CLUSTER_NAME_SEPARATOR = "$";
+  protected static final Logger LOG = LoggerFactory.getLogger(AtlasProcess.class);
+
+  private RESTClientBuilder builder = new RESTClientBuilder();
+  protected AtlasRequestBuilder atlasRequestBuilder = new AtlasRequestBuilder();
+
+  protected AtlasRESTClient getClient(String atlasEndpoint, HiveConf conf) throws SemanticException {
+    if (conf.getBoolVar(HiveConf.ConfVars.HIVE_IN_TEST_REPL)) {
+      return new DummyAtlasRESTClient();
+    }
+    return builder.baseUrl(atlasEndpoint).create();
+  }
+
+  public String getEntityGuid(String atlasEndpoint, String typeName, String attributeName,
+                              String attributeValue, HiveConf conf) throws SemanticException {
+    return getClient(atlasEndpoint, conf).getEntityGuid(typeName, attributeName, attributeValue);
+  }
+
+  public boolean getStatus(String atlasEndpoint, HiveConf conf) throws SemanticException {
+    return getClient(atlasEndpoint, conf).getStatus();
+  }
+
+  public abstract void run(AtlasReplInfo atlasReplInfo) throws SemanticException;
+
+  public long getCurrentTimestamp(AtlasReplInfo atlasReplInfo, String entityGuid) throws SemanticException {
+    AtlasRESTClient client = getClient(atlasReplInfo.getAtlasEndpoint(), atlasReplInfo.getConf());
+    AtlasServer atlasServer = client.getServer(atlasReplInfo.getSrcCluster());
+    long ret = (atlasServer == null || atlasServer.getAdditionalInfoRepl(entityGuid) == null)
+            ? 0L : (long) atlasServer.getAdditionalInfoRepl(entityGuid);
+    LOG.debug("HiveAtlasPlugin: fromTimestamp: {}", ret);
+    return ret;
+  }
+
+  public String checkHiveEntityGuid(String atlasEndpoint, String fullyQualifiedClusterName, String srcDb, HiveConf conf)
+          throws SemanticException {
+    String clusterName = getAtlasClusterName(fullyQualifiedClusterName);
+    AtlasObjectId objectId = atlasRequestBuilder.getItemToExport(clusterName, srcDb);
+    Set<Map.Entry<String, Object>> entries = objectId.getUniqueAttributes().entrySet();
+    if (entries == null || entries.isEmpty()) {
+      throw new SemanticException("HiveAtlasPlugin: Could find entries in objectId for:" + clusterName);
+    }
+    Map.Entry<String, Object> item = entries.iterator().next();
+    String guid = getEntityGuid(atlasEndpoint, objectId.getTypeName(), item.getKey(), (String) item.getValue(), conf);
+    if (guid == null || guid.isEmpty()) {
+      throw new SemanticException("HiveAtlasPlugin: Entity not found:" + objectId);
+    }
+    return guid;
+  }
+
+  protected static String getAtlasClusterName(String clusterName) {
```

Comment:
is this needed?

Reply:
Yes, if orchestration engine adds dc in cluster name. Have retained the behavior.